mingw-w64-firebird/PKGBUILD

```
@@ -0,0 +1,153 @@
+depends=("${MINGW_PACKAGE_PREFIX}-icu"
+         "${MINGW_PACKAGE_PREFIX}-libiconv"
+         "${MINGW_PACKAGE_PREFIX}-libtomcrypt"
+         "${MINGW_PACKAGE_PREFIX}-libtommath"
+         "${MINGW_PACKAGE_PREFIX}-re2")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-autotools"
+             "unzip")
+options=('!buildflags')
+source=("https://github.com/FirebirdSQL/firebird/archive/refs/tags/v${pkgver%.*}.tar.gz"
+        001-configure-fixes.patch
+        002-mingw-makefiles-fixes.patch
+        003-substitute-extauth-makefile.patch
+        004-external-sources-fixes.patch
+        005-firebird-sources-fixes.patch
+        006-resource-file-fixes.patch
+        007-pathtools-fixes.patch
+        008-add-script-and-batch-files.patch
+        pathtools.cpp
+        pathtools.h)
+sha256sums=('4da3aedbbe780f5f878f19998828fa64a9b43285499eff99c5dad71b7c72dcaf'
+            'be70056bebc87a4834ff7785e32915b547b5d62cf14042851d54e5762ea05368'
+            '70b4b5327e128409b9dc2bf14871062a6b36fcffb3e69d30bb6a7f0e36f95b90'
+            'd99dc050c57c43e351000c161d51c0ec0420004de912547fcf04e9325cd6de27'
+            '88959ee452b2950ebaac5e1a034cbb54eb95c886616a322664074704a02ba43e'
+            '84714bc93ba21368b35fe9de757eafbc390742eeb4b5641d2f8330d828b72bba'
+            '76f337487b62c99cff3dd2045f4143a69b0395c82d76ebf42d18d7ca3f5eca66'
+            'f9babbf59dfb99a5d0048647bb64fce39f821c110e768049aa955614396382e2'
+            '094896b8614981854eadcca789726e8eb646918811829f58cde43a27870156ed'
+            '2f357fe2c6cdec3bfd90979616cfd075aee2de51beeaa08cd7a4192878efa64e'
+            '0e6014650703817bc106825bb6a92bea72d9d790e89975c88d058fba2a26eddc')
+
+pkgver() {
+  cd ${_realname}-${pkgver%.*}
+  source ${srcdir}/${_realname}-${pkgver%.*}/src/misc/writeBuildNum.sh
+  printf '%s.%s.%s.%s' ${MajorVer} ${MinorVer} ${RevNo} ${BuildNum}
+}
+
+apply_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying ${_patch}"
+    patch -p1 -i "${srcdir}/${_patch}"
+  done
+}
+
+prepare() {
+  cd ${_realname}-${pkgver%.*}
+  cp -f ../pathtools.h src/common
+  cp -f ../pathtools.cpp src/common/os/win32
+
+  apply_patch_with_msg \
+    001-configure-fixes.patch \
+    002-mingw-makefiles-fixes.patch \
+    003-substitute-extauth-makefile.patch \
+    004-external-sources-fixes.patch \
+    005-firebird-sources-fixes.patch \
+    006-resource-file-fixes.patch \
+    007-pathtools-fixes.patch \
+    008-add-script-and-batch-files.patch
+
+  NOCONFIGURE=1 ./autogen.sh
+}
+
+build() {
+  cd ${_realname}-${pkgver%.*}
+
+  if [[ ${MSYSTEM} == CLANG* ]]; then
+    CXXFLAGS="-Wno-varargs -Wno-constant-conversion -Wno-inconsistent-missing-override -Wno-tautological-constant-out-of-range-compare -Wno-c++11-narrowing"
+  else
+    CXXFLAGS="-Wno-class-memaccess -Wno-overflow -Wno-return-local-addr -Wno-stringop-overread -Wno-stringop-overflow -Wno-narrowing"
+  fi
+
+  ./configure \
```

```suggestion
  mkdir -p build-${MSYSTEM} && cd build-${MSYSTEM}

  if [[ ${MSYSTEM} == CLANG* ]]; then
    CXXFLAGS="-Wno-varargs -Wno-constant-conversion -Wno-inconsistent-missing-override -Wno-tautological-constant-out-of-range-compare -Wno-c++11-narrowing"
  else
    CXXFLAGS="-Wno-class-memaccess -Wno-overflow -Wno-return-local-addr -Wno-stringop-overread -Wno-stringop-overflow -Wno-narrowing"
  fi

  ../${_realname}-${pkgver%.*}/configure \
```
Is out-of-tree building possible?

---

tried once before.. and it failed (something related to the generated source files)